vyper/semantics/types/function.py

```diff
@@ -377,6 +382,9 @@ def from_FunctionDef(cls, funcdef: vy_ast.FunctionDef) -> "ContractFunctionT":
         """
         function_visibility, state_mutability, nonreentrant = _parse_decorators(funcdef)
 
+        # it's redundant to specify internal visibility - it's implied by not being external
+        function_visibility = function_visibility or FunctionVisibility.INTERNAL
```

Comment:
```suggestion
        if function_visibility is None:
            FunctionVisibility.INTERNAL
```

Reply:
addressed in: a6f1a86ae7f3a6c920682e1b30c8675244b4a504